test/healthcheck_test.go

```diff
@@ -53,15 +53,51 @@ func replaceHealthCheckFunc(f func(context.Context, func() (interface{}, error),
 
 func newTestHealthServer() *testHealthServer {
 	return &testHealthServer{
-		update: make(chan struct{}, 1),
-		status: make(map[string]healthpb.HealthCheckResponse_ServingStatus),
+		watchFunc: defaultWatchFunc,
+		update:    make(chan struct{}, 1),
+		status:    make(map[string]healthpb.HealthCheckResponse_ServingStatus),
 	}
 }
 
+func newTestHealthServerWithSpecialWatchFunc(f func(s *testHealthServer, in *healthpb.HealthCheckRequest, stream healthpb.Health_WatchServer) error) *testHealthServer {
```

Comment:
s/special//

Or delete the non-parameterized form and just let the initializers explicitly set "defaultWatchFunc".